packages/shared/src/components/cards/WelcomePostCardFooter.tsx

```diff
@@ -11,17 +11,32 @@ type WelcomePostCardFooterProps = {
 export const WelcomePostCardFooter = ({
   post,
 }: WelcomePostCardFooterProps): ReactElement => {
+  const [image, setImage] = useState(post?.image);
   const content = useMemo(
     () =>
       post?.contentHtml ? sanitize(post.contentHtml, { ALLOWED_TAGS: [] }) : '',
     [post?.contentHtml],
   );
 
-  if (post.image) {
+  const imageCheck = useMemo(() => {
+    if (image) return true;
+
+    const parser = new DOMParser();
+    const doc = parser.parseFromString(post?.contentHtml, 'text/html');
+    const imgTag = doc.querySelector('img');
+    if (imgTag) {
+      setImage(imgTag.getAttribute('src') || null);
```

Comment:
You don't need state here, state setting from `useMemo` is anti-pattern because it is the same as setting state during rendering.

You can just return the `image` from `useMemo` and use it below for boolean check.

Reply:
Very good point, iv'e updated it to remove the useState and just use useMemo instead